modules/qna-openai/additional/answer/answer_result.go

```diff
@@ -0,0 +1,106 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package answer
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/semi-technologies/weaviate/entities/models"
+	"github.com/semi-technologies/weaviate/entities/moduletools"
+	"github.com/semi-technologies/weaviate/entities/search"
+	qnamodels "github.com/semi-technologies/weaviate/modules/qna-openai/additional/models"
+)
+
+func (p *AnswerProvider) findAnswer(ctx context.Context, in []search.Result, params *Params, limit *int, argumentModuleParams map[string]interface{}, cfg moduletools.ClassConfig) ([]search.Result, error) {
+	if len(in) > 0 {
+		question := p.paramsHelper.GetQuestion(argumentModuleParams["ask"])
+		if question == "" {
+			return in, errors.New("empty question")
+		}
+		properties := p.paramsHelper.GetProperties(argumentModuleParams["ask"])
+
+		for i := range in {
+			textProperties := map[string]string{}
+			schema := in[i].Object().Properties.(map[string]interface{})
```

Comment:
I'm not sure what would be the best way to change it without the casting.

Reply:
```
schema, ok := in[i].Object().Properties.(map[string]interface{})
if !ok {
   return errors.New("...")
}
```

not sure if that is necessary, I think it would complain way earlier if the properties would not be a map